packages/database/src/models/fhir/FhirSpecimen.ts

```diff
@@ -28,6 +28,10 @@ export class FhirSpecimen extends FhirResource {
 
     this.UpstreamModels = [models.LabRequest];
     this.upstreams = [models.LabRequest];
+    this.referencedResources = [
+      models.FhirPractitioner,
+      // FhirServiceRequest is also referenced by Specimen, but not flagging here to avoid a circular dependency
```

Comment:
This feels really quite awkward (similarly in FhirPatient). But it's a bit hard to avoid. FHIR contains circular dependencies by design, so we need to have some way of resolving despite this.

I considered correctly recording all the referenced resources here and then adding a separate field/function to filter those resources to avoid circular dependencies. But given the only use case for this field at the moment is sorting by dependency order I decided it's fine to just omit them here and leave a comment to avoid any confusion. 

If something else pops up in the future that requires the full list of referenced resources then we may need to change this.